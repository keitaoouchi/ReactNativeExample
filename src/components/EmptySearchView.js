import React from "react";
import {Text, TouchableWithoutFeedback, View} from "react-native";
import Workaround from "../misc/Workaround";
import Color from "../misc/Color";
import dismissKeyboard from "dismissKeyboard";

export default EmptySearchView = () => (
  <TouchableWithoutFeedback onPress={dismissKeyboard}>
    <View style={style.root}>
      <Text style={style.mainText}>該当する結果は見つかりませんでした</Text>
      <Text style={style.subText}>別のキーワードをお試しください</Text>
    </View>
  </TouchableWithoutFeedback>
);

const style = {
  root: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: Color.black,
    ...Workaround.root,
  },
  mainText: {
    color: 'white',
    fontSize: 20,
    margin: 8,
  },
  subText: {
    color: 'white',
    opacity: 0.7,
    fontSize: 14,
    marginLeft: 16,
    marginRight: 16,
  },
};