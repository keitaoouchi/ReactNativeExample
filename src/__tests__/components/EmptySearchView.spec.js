import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import EmptySearchView from "../../../src/components/EmptySearchView";

describe('EmptySearchView', () => {

  it('renders correctly', () => {
    const tree = renderer.create(
      <EmptySearchView />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});