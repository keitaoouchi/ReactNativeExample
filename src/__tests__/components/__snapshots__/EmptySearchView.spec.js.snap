// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptySearchView renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#181818",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
  testID={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 20,
        "margin": 8,
      }
    }
  >
    該当する結果は見つかりませんでした
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 14,
        "marginLeft": 16,
        "marginRight": 16,
        "opacity": 0.7,
      }
    }
  >
    別のキーワードをお試しください
  </Text>
</View>
`;
